llvm/test/CodeGen/AArch64/dag-preserve-icmp-samesign.ll

```diff
@@ -0,0 +1,31 @@
+; NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py UTC_ARGS: --version 5
+; RUN: llc -march=aarch64 -stop-after=finalize-isel -simplify-mir -o - %s | FileCheck %s
+
+define i1 @icmp_samesign(i32 %a, i32 %b) {
+  ; CHECK-LABEL: name: icmp_samesign
+  ; CHECK: bb.0 (%ir-block.0):
+  ; CHECK-NEXT:   liveins: $w0, $w1
+  ; CHECK-NEXT: {{  $}}
+  ; CHECK-NEXT:   [[COPY:%[0-9]+]]:gpr32 = COPY $w1
+  ; CHECK-NEXT:   [[COPY1:%[0-9]+]]:gpr32 = COPY $w0
+  ; CHECK-NEXT:   [[SUBSWrr:%[0-9]+]]:gpr32 = SUBSWrr [[COPY1]], [[COPY]], implicit-def $nzcv
```

Comment:
@arsenm Was trying to test DAG preserving of the flag (to test InstrEmitter), but it seems like the flag gets lost during legalization of the DAG (which boils down to a custom LowerSETCC for AArch64/X86). Shouldn't we want to preserve the flag there as well, or is it beyond patch scope?

Reply:
The value of preserving the flag is diminishing the further back you go. I doubt anything after selection will have any use of this

Reply:
Shouldn't we still preserve the flag immediately before doing instruction selection (before the very last iteration of DAG combining)?

Reply:
Don't really care, it's easy enough to handle. I would agree preserving the flag in other contexts is beyond the scope of this patch

Reply:
Okay, I'm dropping the test then, and merging this, if you are still fine with everything else.